src/CoreServices/FSEvents.cs

```diff
@@ -195,11 +278,32 @@ protected override void Release ()
 			ref FSEventStreamContext context, IntPtr pathsToWatch,
 			ulong sinceWhen, double latency, FSEventStreamCreateFlags flags);
 
-		public FSEventStream (CFAllocator? allocator, NSArray pathsToWatch,
-			ulong sinceWhenId, TimeSpan latency, FSEventStreamCreateFlags flags)
+		[DllImport (Constants.CoreServicesLibrary)]
+		unsafe static extern IntPtr FSEventStreamCreateRelativeToDevice (IntPtr allocator,
+#if NET
+			delegate* unmanaged<IntPtr, IntPtr, nint, IntPtr, IntPtr, IntPtr, void> callback,
+#else
+			FSEventStreamCallback callback,
+#endif
+			ref FSEventStreamContext context, ulong deviceToWatch, IntPtr pathsToWatchRelativeToDevice,
+			ulong sinceWhen, double latency, FSEventStreamCreateFlags flags);
+
+		public FSEventStream (FSEventStreamCreateOptions options)
 		{
-			if (pathsToWatch is null)
-				throw new ArgumentNullException (nameof (pathsToWatch));
+			if (options is null)
+				throw new ArgumentNullException (nameof (options));
+
+			NSArray pathsToWatch;
+
+			if (options.NSPathsToWatch is not null)
```

Comment:
Nit: We tends to use braces for blocks like this, specially the long throw case.